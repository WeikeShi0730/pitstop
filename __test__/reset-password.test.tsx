import { render, RenderResult } from "@testing-library/react";
import RestPassword from "../components/reset-password.component";

let documentBody: RenderResult;
describe("<RestPassword />", () => {
  beforeEach(() => (documentBody = render(<RestPassword />)));
  it("Test <RestPassword />", () => {
    expect(
      documentBody.queryByText("Send a password reset email")
    ).toBeInTheDocument();
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
