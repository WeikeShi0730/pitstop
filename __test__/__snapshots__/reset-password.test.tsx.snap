// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RestPassword /> Test <RestPassword /> 1`] = `
<body>
  <div>
    <div
      class="min-h-content flex justify-center items-center"
    >
      <div
        class="max-w-xs sm:max-w-sm md:max-w-md lg:max-w-lg m-auto my-10 text-slate-700 bg-slate-100 bg-opacity-30 rounded-lg py-8 px-10 shadow-md hover:shadow-slate-500 transition-all ease-in-out duration-200 font-light"
      >
        <div
          class="text-center m-5 text-base lg:text-xl"
        >
          Reset my password
        </div>
        <form>
          <div>
            <label
              class="text-sm md:text-base"
            >
              Email
            </label>
            <input
              class="text-sm md:text-base w-full p-2 border-b border-slate-700 outline-none bg-transparent mb-4 rounded-none"
              id="email"
              name="email"
              placeholder="Your email"
              required=""
              type="email"
            />
          </div>
          <div
            class="flex justify-center items-center mt-6"
          >
            <button
              class="text-sm md:text-base bg-slate-700 py-2 px-4 text-slate-100 rounded outline-none font-light hover:shadow-md hover:shadow-slate-700 hover:bg-slate-500 hover:text-white transition-all ease-in-out duration-200"
              type="submit"
            >
              Send a password reset email
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</body>
`;
